### PR TITLE
FreezedWeights has been added

### PR DESCRIPTION
**Pull request by tuv-mohamad-atayi**
VidSal model has been split from CNN3d.
The option to freeze weights of different layers has been added to the model.

### CODE REVIEW

File: model/model.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import torch.nn.functional as F
+import torchvision as tv
+from torch.autograd import Variable
+import math
+from model.mdn import MDN1D
+from model.cnn3d import CNN3d
+
+class VidSal(nn.Module):
+    def __init__(self, model_path, hidden_size=256, LSTM_layers=1, freeze_cnn3d_weights =[True,True,True,True,True]):
+        super(CNN3d, self).__init__()
```
Comment (AAshqar) [inline model/model.py]: "super(VidSal, self).__init__()" instead of "super(CNN3d, self).__init__()"

File: model/model.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        
+        self.hidden_size = hidden_size
+        self.LSTM_layers = LSTM_layers
+        
+        ## To 
+        self.cnn3d = CNN3d(model_path,freeze_weights = freeze_cnn3d_weights)
```
Comment (AAshqar) [inline model/model.py:21]: CNN3d(model_path, freeze_cnn3d_weights = freeze_cnn3d_weights)

File: model/model.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        return x
+    
+    def init_hidden(self):
+        # the first is the hidden h
+        # the second is the cell  c
+        #(num_layers * num_directions, batch, hidden_size): tensor containing the hidden state for t=seq_len
+        return (Variable(torch.zeros(self.LSTM_layers, 1, self.hidden_size)),
+                Variable(torch.zeros(self.LSTM_layers, 1, self.hidden_size)))
+    
+        @property
```
Comment (AAshqar) [inline model/model.py]: @property should have one less indentation

**Review by AAshqar — CHANGES_REQUESTED** [review]
model file (see comments)

File: model/model.py (hunk 1 above)
Comment (tuv-mohamad-atayi) [reply model/model.py]: Right!

**Review by AAshqar — APPROVED** [review]

**Comment by AAshqar** [comment]
Great! :D now it's working perfectly

**Pull request merged**